### DISCUSSION TITLE
check the present of Xcode before installing Command Line Tools

### DISCUSSION THREAD

**Discussion by leungyauming** [Homebrew/discussions#2335] [Getting started] [answered]
According to https://docs.brew.sh/Installation#3,
<img width="641" alt="Screenshot 2021-10-23 at 1 51 50 PM" src="https://user-images.githubusercontent.com/50203211/138544320-f2730fa2-c617-4154-aa61-fdd92d6a87f9.png">
`You can install Xcode, the CLT, or both`, but the current Homebrew installer scripts **ONLY** check the present of Command Line Tools (https://github.com/Homebrew/install/blob/master/install.sh#L251).

I believe checking the present of Xcode is reasonable and can reduce changes to users' development environment.

**Comment by SMillerDev** [comment, marked as answer]
I believe the documentation is wrong because apple started making changes to the CLT. It's no longer shipping the same tools as xcode. Could you make a pull request to edit the documentation?

**Reply by leungyauming** [reply]
Just to confirm, does that mean Xcode is **NOT** supported and the CLT should be used instead?

**Reply by carlocab** [reply]
We do still support all three configurations. There's a handful of formulae whose bottles require a CLT installation, though, so the install script checks for the CLT. Using the Xcode-only configuration requires using an alternative installation method.

**Reply by SMillerDev** [reply]
It means that some software needs xcode, and it'll tell you when it does. Most software need the commandline tools though. So that's important to install first.

**Reply by leungyauming** [reply]
Then which one should I use as the active developer directory?

**Reply by leungyauming** [reply]
@carlocab Could you please tell me more about the Xcode-only installation method?

**Reply by carlocab** [reply]
> Then which one should I use as the active developer directory?

If you have the CLT installed, use the CLT.

> @carlocab Could you please tell me more about the Xcode-only installation method?

It's in the [documentation](https://docs.brew.sh/Installation#untar-anywhere).
